packages/w3p/src/providers/raw/near-raw-provider.test.ts

```diff
@@ -0,0 +1,135 @@
+import { setupWalletSelector, WalletSelector } from '@near-wallet-selector/core'
+import type { Options } from '@near-wallet-selector/core/lib/options.types'
+import { Observable } from 'rxjs'
+
+import { NEAR_CHAINS } from '@/enums'
+import { NearRawProvider } from '@/providers'
+
+jest.mock('@near-wallet-selector/core', () => ({
+  setupWalletSelector: jest.fn(() => ({
+    isSignedIn: jest.fn().mockReturnValue(false),
+    store: {
+      getState: jest.fn().mockReturnValue({
+        accounts: [{ accountId: 'test_account' }],
+      }),
+    },
+    wallet: jest.fn().mockReturnValue({
+      signIn: jest.fn(),
+      signOut: jest.fn(),
+      signAndSendTransactions: jest.fn(),
+    }),
+  })),
+  setupMyNearWallet: jest.fn(),
+}))
+let mockWallet: {
+  signIn: jest.Mock
+  signOut: jest.Mock
+  signAndSendTransactions: jest.Mock
+}
+describe('performs NearRawProvider unit testing', () => {
+  let provider: NearRawProvider
+  beforeEach(() => {
+    provider = new NearRawProvider({ createAccessKeyFor: 'test' })
+    mockWallet = {
+      signIn: jest.fn(),
+      signOut: jest.fn(),
+      signAndSendTransactions: jest.fn(),
+    }
+  })
+  afterEach(() => {
+    jest.restoreAllMocks()
+  })
+
+  describe('performs init', () => {
+    test('should setup wallet selector and check isSignedIn', async () => {
+      const mockSelector: WalletSelector = {
+        isSignedIn: jest.fn().mockReturnValue(true),
+        wallet: jest.fn().mockResolvedValue(mockWallet),
+        store: {
+          getState: jest.fn().mockReturnValue({
+            accounts: [{ accountId: 'test' }],
+          }),
+          observable: new Observable(),
+        },
+        options: {} as Options,
+        setActiveAccount: jest.fn,
+        on: jest.fn(),
+        off: jest.fn(),
+      }
+      ;(
+        setupWalletSelector as jest.MockedFunction<typeof setupWalletSelector>
+      ).mockResolvedValue(mockSelector)
+      const result = await provider.init()
```

Comment:
rename `result` variable and why `provider.init()` returning something?

Reply:
renamed
init returned boolean value, whether the initialization was successful

Reply:
need to discuss